webpack.config.js

```diff
@@ -1,36 +1,145 @@
-var uglifyJS = require('uglify-es');
-var fs = require('fs');
+const path = require('path');
+const webpack = require('webpack');
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');
+const CopyWebpackPlugin = require('copy-webpack-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const uglifyJS = require('uglify-es');
+const fs = require('fs');
 
-var isDev = (process.env.NODE_ENV === 'development');
-// these values are required in the config.app.js file -- we can't use
-// process.env with webpack, we have to create these magic constants
-// individually.
-var defineEnvPlugin = new webpack.DefinePlugin({
-  'process.env': {
-    'NODE_ENV': isDev ? JSON.stringify('development') : JSON.stringify('production')
+const isDev = (process.env.NODE_ENV === 'development');
+const isTest = (process.env.NODE_ENV === 'test');
+
+// Enzyme as of v2.4.1 has trouble with classes
+// that do not start and *end* with an alpha character
+// but that will sometimes happen with the base64 hashes
+// so we leave them off in the test env
+const localIdentName = process.env.NODE_ENV === 'test'
+  ? '[name]--[local]'
+  : '[name]--[local]--[hash:base64:5]';
+
+const styleLoaderConfiguration = {
+  test: /\.less$/,
+  use: [
+    (isDev || isTest) ? 'style-loader' : MiniCssExtractPlugin.loader,
+    {
+      loader: 'css-loader',
+      query: {
+        importLoaders: 2,
+        localIdentName,
+        sourceMap: true,
+      },
+    },
+    {
+      loader: 'postcss-loader',
+      options: {
+        sourceMap: true,
+      },
+    },
+    {
+      loader: 'less-loader',
+      options: {
+        sourceMap: true,
+        javascriptEnabled: true,
+      },
+    },
+  ],
+};
+
+const babelLoaderConfiguration = [
+  {
+    test: /\.js$/,
+    // exclude: /node_modules/,
+    exclude: function(modulePath) {
+      return /node_modules/.test(modulePath) && !/node_modules\/(tideline)/.test(modulePath);
+    },
+    use: {
+      loader: 'babel-loader',
+      options: {
+        cacheDirectory: true,
+      },
+    },
   },
-  __UPLOAD_API__: JSON.stringify(process.env.UPLOAD_API || null),
-  __API_HOST__: JSON.stringify(process.env.API_HOST || null),
-  __INVITE_KEY__: JSON.stringify(process.env.INVITE_KEY || null),
-  __LATEST_TERMS__: JSON.stringify(process.env.LATEST_TERMS || null),
-  __PASSWORD_MIN_LENGTH__: JSON.stringify(process.env.PASSWORD_MIN_LENGTH || null),
-  __PASSWORD_MAX_LENGTH__: JSON.stringify(process.env.PASSWORD_MAX_LENGTH || null),
-  __ABOUT_MAX_LENGTH__: JSON.stringify(process.env.ABOUT_MAX_LENGTH || null),
-  __I18N_ENABLED__: JSON.stringify(process.env.I18N_ENABLED || false),
-  __DEV__: isDev,
-  __TEST__: false,
-  __DEV_TOOLS__: (process.env.DEV_TOOLS != null) ? process.env.DEV_TOOLS : (isDev ? true : false) //eslint-disable-line eqeqeq
-});
+  // {
```

Comment:
Is this comment block useful to leave in?

Reply:
No, I just missed cleaning it up once I got the new config working.  Thanks 👍